File: public/js/components/discussion-board/proposal.js
var Proposal = function(id) {

    var my_id = id;
    var title = document.getElementById('title');
    var content = document.getElementById('content');
    var circle = document.getElementById('circle');

    var render = function(data) {
        title.innerHTML = data.title;
        content.innerHTML = data.content;
        circleContent = populateCircle(data.circle);
        circle.append(circleContent);
    };

    var createListItem = function(username) {
        listItem = document.createElement('li');
        listItem.className = 'user-involved list-group-item';
        listItem.innerHTML = username;
        return listItem;
    };

    var populateCircle = function(circle) {
        var listItems = document.createDocumentFragment();
        circle.forEach(function(username) {
            listItems.append(createListItem(username));
        });
        return listItems;
    };

    Bus.subscribe('proposal.retrieved', render);
    Bus.publish('proposal.retrieve', my_id);

};

File: domain/proposal.rb
require_relative 'circle'

module Domain
  class Proposal
    def initialize(id, proposer)
      @id = id
      @circle = Circle.new
      @proposer = proposer
      involve(@proposer)
    end

    def attach(title, content)
      @title = title
      @content = content
    end

    def involve(person)
      @circle.involve(person)
    end

    def involved
      @circle.involved
    end

    def involved?(person)
      @circle.involved?(person)
    end
  end
end
